From: Simon McVittie <smcv@debian.org>
Date: Sun, 22 Dec 2019 21:34:45 +0000
Subject: tests: Skip Location tests if we have no D-Bus system bus

We can't expect Geoclue to work in that situation. Ideally the location
portal would report this as an error, but at the moment the test just
hangs if we cannot connect to the system bus.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/location.c | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/tests/location.c b/tests/location.c
index 0395f77..e1eacb3 100644
--- a/tests/location.c
+++ b/tests/location.c
@@ -29,12 +29,23 @@ void
 test_location_basic (void)
 {
   g_autoptr(XdpPortal) portal = NULL;
+  g_autoptr(GDBusConnection) system_bus = NULL;
+  g_autoptr(GError) error = NULL;
 
 #ifndef HAVE_GEOCLUE
   g_test_skip ("Skipping tests that require geoclue");
   return;
 #endif
 
+  system_bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, &error);
+
+  if (system_bus == NULL)
+    {
+      g_prefix_error (&error, "Unable to test Location without system bus: ");
+      g_test_skip (error->message);
+      return;
+    }
+
   portal = xdp_portal_new ();
 
   got_info = 0;
@@ -66,12 +77,23 @@ void
 test_location_accuracy (void)
 {
   g_autoptr(XdpPortal) portal = NULL;
+  g_autoptr(GDBusConnection) system_bus = NULL;
+  g_autoptr(GError) error = NULL;
 
 #ifndef HAVE_GEOCLUE
   g_test_skip ("Skipping tests that require geoclue");
   return;
 #endif
 
+  system_bus = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, &error);
+
+  if (system_bus == NULL)
+    {
+      g_prefix_error (&error, "Unable to test Location without system bus: ");
+      g_test_skip (error->message);
+      return;
+    }
+
   portal = xdp_portal_new ();
 
   got_info = 0;
